framework/db/BaseActiveRecord.php

```diff
@@ -1781,4 +1781,57 @@ private function isValueDifferent($newValue, $oldValue)
 
         return $newValue !== $oldValue;
     }
+
+    /**
+     * Eager loads related models for the already loaded primary models.
+     *
+     * Helps to reduce the number of queries performed against database if some related models are only used
+     * when a specific condition is met. For example:
+     *
+     * ```php
+     * $customers = Customer::find()->where(['country_id' => 123])->all();
+     * if (Yii:app()->getUser()->getIdentity()->canAccessOrders()) {
+     *     Customer::loadRelationsFor($customers, 'orders.items');
+     * }
+     * ```
+     *
+     * @param array|ActiveRecordInterface[] $models array of primary models. Each model should have the same type and can be:
+     * - an active record instance;
+     * - active record instance represented by array (i.e. active record was loaded using [[ActiveQuery::asArray()]]).
+     * @param string|array $relationNames the names of the relations of primary models to be loaded from database. See [[ActiveQueryInterface::with()]] on how to specify this argument.
+     * @param bool $asArray whether to load each related model as an array or an object (if the relation itself does not specify that).
+     * @since 2.0.49
+     */
+    public static function loadRelationsFor(&$models, $relationNames, $asArray = false)
+    {
+        // ActiveQueryTrait::findWith() called below assumes $models array is non-empty.
+        if (empty($models)) {
```

Comment:
Why not ` if (!empty($models))` and avoid empty `return` statement which raises Codecov warning?

Reply:
It will make the code less readable IMO:
- your variant uses extra logical operator (which makes expression more complex for human understanding);
- the ! operator is easily missable at a glance;
- my variant has a clearly visible useful work of a function (last line) and a clearly separated condition when it can be skipped; your variant would have these tangled up making it harder to understand what is the normal function flow and what are the exceptions in it.

Making the code less readable just to shut up a warning in some imperfect external tool is a bad idea.

Reply:
But shouldn't it cover the test, for when the model is empty?

Reply:
> But shouldn't it cover the test, for when the model is empty?

You mean if there should be a test for a `loadRelationsFor([], /*...*/)` call? I don't think this test will be very useful.

Reply:
But it is testing the code, in the case that the model is empty.

Reply:
> But it is testing the code, in the case that the model is empty.

So you just want to do the testing for the sake of doing testing even if it doesn't test anything useful at all? If you disagree, please show us how such test could possibly look and explain its usefulness.

Reply:
@PowerGamer1 when things breaks for empty models with some bug in the code (let say someone else modified your now empty method to something else, how will we detect it? That's where this test comes to rescue. Currently sound like useless, but it will make code "future proof" sort of!

Reply:
> @PowerGamer1 ... let say someone else modified your now empty return to something else, how will we detect it?

At present, testing a single return statement is useless. If, in the future, as you say someone replaces/modifies this return statement with a MORE COMPLEX piece of code - then at that point in time testing such MORE COMPLEX piece of code might become warranted. Obviously, testing of such (currently non-existent) more complex code path should be done when it is introduced by the person who introduced it.

Also, if I were you (i.e. engine maintainer), I would be much more concerned with testing infinitely more complex and currently completely untested pieces of code (such as `ActiveQueryTrait::findWith()` and myriad others). In fact, the only reason I am currently writing any tests at all is the fact that new shortcut methods rely on such untested functionality - i.e. I don't really test NEW code, I have to test already EXISTING code.

P.S. That test you want me to write - have you even imagined how exactly it will look? What exactly are you going to assert (like `assertTrue()` etc.) in it?